src/components/Payroll/RunPayrollFlow/RunPayrollFlow.stories.tsx

```diff
@@ -0,0 +1,406 @@
+import {
+  createContext,
+  use,
+  useContext,
+  useReducer,
+  useState,
+  type ActionDispatch,
+  type ReactNode,
+} from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
+import { action } from '@ladle/react'
+import { DataView, Flex, NumberInputField } from '@/components/Common'
+import { useComponentContext } from '@/contexts/ComponentAdapter/useComponentContext'
+import { HamburgerMenu } from '@/components/Common/HamburgerMenu'
+import PencilSvg from '@/assets/icons/pencil.svg?react'
+import { Form } from '@/components/Common/Form'
+import { BaseComponent } from '@/components/Base'
+
+export default {
+  title: 'Domain/Run Payroll/Flow',
+}
+
+// **** PRESENTATIONAL ****
+// **** PRESENTATIONAL ****
+// **** PRESENTATIONAL ****
+
+interface PayrollListProps {
+  onRunPayroll: ({ payrollId }: { payrollId: string }) => void
+  payrolls: any[]
+}
+const PayrollList = ({ onRunPayroll, payrolls }: PayrollListProps) => {
```

Comment:
I'm assuming this one would have a parent PayrollListBlock as well that is handling onRunPayroll and fetching the payrolls?

Reply:
@serikjensen  This is part of the question I was asking re: are the blocks in the right place. Assume for now that the things named Block are the blocks we are shipping and everything else is only reachable via the Flow. What do you think about these choices? Do we have sufficient evidence that everything should be a block even if their available actions don't make as much sense without the rest of the blocks?

Reply:
yeah my thought in still keeping them as blocks would be
* Consistency with the development approach
* Ability to integrate with an application router
* Flexibility for the partner (not a super strong argument in this case, unlikely they would be moving things around)

I'd say i care about things in that order. Keeping our development approach consistent across the domains for ease of understanding and knowing where things are as well as unlocking the ability to manage the transitions with a router instead of a state machine as outlined here https://docs.gusto.com/embedded-payroll/docs/routing